test_runner/random_test_runner/src/lanelet_utils.cpp

```diff
@@ -0,0 +1,292 @@
+{
+  return hdmap_utils_ptr_->toMapPose(lanelet_pose);
+}
+
+std::vector<std::int64_t> LaneletUtils::getRoute(
+  std::int64_t from_lanelet_id, std::int64_t to_lanelet_id)
+{
+  return hdmap_utils_ptr_->getRoute(from_lanelet_id, to_lanelet_id);
+}
+
+double LaneletUtils::getLaneletLength(std::int64_t lanelet_id)
+{
+  return hdmap_utils_ptr_->getLaneletLength(lanelet_id);
+}
+
+double LaneletUtils::computeDistance(
+  traffic_simulator_msgs::msg::LaneletPose p1, traffic_simulator_msgs::msg::LaneletPose p2)
+{
+  auto p1_g = hdmap_utils_ptr_->toMapPose(p1).pose.position;
+  auto p2_g = hdmap_utils_ptr_->toMapPose(p2).pose.position;
+  geometry_msgs::msg::Point d;
+  d.x = p1_g.x - p2_g.x;
+  d.y = p1_g.y - p2_g.y;
+  d.z = p1_g.z - p2_g.z;
+  return std::sqrt(d.x * d.x + d.y * d.y + d.z * d.z);
+}
+
+bool LaneletUtils::isInLanelet(std::int64_t lanelet_id, double s)
+{
+  return hdmap_utils_ptr_->isInLanelet(lanelet_id, s);
+}
+
+boost::optional<traffic_simulator_msgs::msg::LaneletPose> LaneletUtils::getOppositeLaneLet(
+  const traffic_simulator_msgs::msg::LaneletPose & pose)
+{
+  // Assumptions:
+  // - chosen lane is right most and close to opposite one
+  // - traffic is left-hand
+  // - lane width is approximately constant on all the length
+
+  // TODO: Multiple same-direction lane support
+  // TODO: Find lane width for current s value
+
+  if (!lanelet_map_ptr_->laneletLayer.exists(pose.lanelet_id)) {
+    return {};
+  }
+
+  auto tangent_vector = hdmap_utils_ptr_->getTangentVector(pose.lanelet_id, pose.s);
+  if (!tangent_vector) {
+    return {};
+  }
+  lanelet::ConstLanelet current_lanelet = lanelet_map_ptr_->laneletLayer.get(pose.lanelet_id);
+  auto left_point = current_lanelet.leftBound().front();
+  auto right_point = current_lanelet.rightBound().front();
+
+  double lane_width = lanelet::geometry::distance3d(left_point, right_point);
+
+  if (!vehicle_routing_graph_ptr_->rights(current_lanelet).empty()) {
+    // unsupported situation
+    return {};
+  }
+
+  geometry_msgs::msg::Vector3 perpendicular_vector;
+  perpendicular_vector.x = tangent_vector->y;
+  perpendicular_vector.y = -tangent_vector->x;
+  perpendicular_vector.z = 0.0;
+  perpendicular_vector = traffic_simulator::math::normalize(perpendicular_vector);
+
+  geometry_msgs::msg::Point global_position_p = toMapPose(pose).pose.position;
+  geometry_msgs::msg::Vector3 global_position;
+  global_position.x = global_position_p.x;
+  global_position.y = global_position_p.y;
+  global_position.z = global_position_p.z;
+
+  geometry_msgs::msg::Vector3 opposite_lane_global_position;
+  opposite_lane_global_position = global_position + perpendicular_vector * lane_width;
+
+  geometry_msgs::msg::Pose global_pose;
+  global_pose.position.x = opposite_lane_global_position.x;
+  global_pose.position.y = opposite_lane_global_position.y;
+  global_pose.position.z = opposite_lane_global_position.z;
+
+  auto opposite_lanelet_pose = hdmap_utils_ptr_->toLaneletPose(global_pose);
```

Comment:
Can't you just `return hdmap_utils_ptr_->toLaneletPose(global_pose);`? `toLaneletPose` returns `boost::optional<traffic_simulator_msgs::msg::LaneletPose>` anyway